src/frontend/qt_sdl/main.cpp

```diff
@@ -2188,101 +2287,128 @@ bool MainWindow::preloadROMs(QStringList file, QStringList gbafile, bool boot)
     return true;
 }
 
-QString MainWindow::pickFileFromArchive(QString archiveFileName)
+QStringList MainWindow::splitArchivePath(const QString& filename, bool useMemberSyntax)
 {
-    QVector<QString> archiveROMList = Archive::ListArchive(archiveFileName);
+    if (filename.isEmpty()) return {};
 
-    QString romFileName = ""; // file name inside archive
-
-    if (archiveROMList.size() > 2)
+#ifdef ARCHIVE_SUPPORT_ENABLED
+    if (useMemberSyntax)
     {
-        archiveROMList.removeFirst();
+        printf("Warning: use the a.zip|b.nds format at your own risk!\n");
 
-        bool ok;
-        QString toLoad = QInputDialog::getItem(this, "melonDS",
-                                  "This archive contains multiple files. Select which ROM you want to load.", archiveROMList.toList(), 0, false, &ok);
-        if (!ok) // User clicked on cancel
-            return QString();
+        const QStringList filenameParts = filename.split('|');
+        if (filenameParts.size() > 2)
+        {
+            QMessageBox::warning(this, "melonDS", "This path contains too many vertical bars.");
```

Comment:
I'd rather write "This path contains too many '|'."

Reply:
Fixed in 0822c2f8b20eeca6a4889b0a82eceb518c71a761.